Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -330,44 +330,34 @@ public final class RegistryClient {
         )
 
         self.httpClient.execute(request, observabilityScope: observabilityScope, progress: nil) { result in
-            switch result {
-            case .success(let response):
-                do {
-                    try self.checkResponseStatusAndHeaders(response, expectedStatusCode: 200, expectedContentType: .json)
-
-                    guard let data = response.body else {
-                        throw RegistryError.invalidResponse
-                    }
+            _ = result.tryMap { response in
```

Comment:
`completion(result.tryMap` otherwise we will be dropping the errors?

Reply:
actually we may not be able to use tryMap here because of the second nesting so the original code may have the correct way

Reply:
switching back to do-catch. I don't think we can use `tryMap` here since the callback is in nested code.